src/Elf.zig

```diff
@@ -1963,6 +1972,37 @@ fn validateOrSetCpuArch(self: *Elf, name: []const u8, cpu_arch: std.Target.Cpu.A
             @tagName(cpu_arch.toElfMachine()),
             @tagName(self_cpu_arch.toElfMachine()),
         });
+        return error.MismatchedCpuArch;
+    }
+}
+
+fn validateEFlags(self: *Elf, name: []const u8, e_flags: elf.Elf64_Word) !void {
+    // validateOrSetCpuArch should be called before this.
+    const self_cpu_arch = self.options.cpu_arch.?;
+
+    if (self.first_eflags == null) self.first_eflags = e_flags;
+    const self_eflags: *elf.Elf64_Word = &self.first_eflags.?;
+
+    switch (self_cpu_arch) {
+        .riscv64 => {
+            if (e_flags != self_eflags.*) {
+                const riscv_eflags: riscv.RiscvEflags = @bitCast(e_flags);
+                const self_riscv_eflags: *riscv.RiscvEflags = @ptrCast(self_eflags);
+
+                self_riscv_eflags.rvc = self_riscv_eflags.rvc or riscv_eflags.rvc;
+                self_riscv_eflags.tso = self_riscv_eflags.tso or riscv_eflags.tso;
+
+                if (self_riscv_eflags.fabi != riscv_eflags.fabi) {
+                    self.base.fatal("{s}: cannot link object files with different float-point ABIs", .{name});
+                }
+                if (self_riscv_eflags.rve != riscv_eflags.rve) {
+                    self.base.fatal("{s}: cannot link object files with different RVEs", .{name});
+                }
+
+                return error.MismatchedEflags;
```

Comment:
Is it possible for `e_flags != self_eflags.*` while `self_riscv_eflags.fabi == riscv_eflags.fabi` and `self_riscv_eflags.rve == riscv_eflags.rve`? If so we should report a generic error that flags differ as otherwise we will return `error.MismatchedEflags` with no user error.

Reply:
Oh, that's totally my fault. It is possible, I meant to create an `is_error` flag of sorts and return if it's true.

Reply:
Fixed using a flag to track if an error happened.